src/devices/Vl53L1X/Vl53L1X.cs

```diff
@@ -0,0 +1,976 @@
+        /// </summary>
+        public ushort Xtalk
+        {
+            get
+            {
+                ushort xTalk = ReadUInt16((ushort)Registers.ALGO__CROSSTALK_COMPENSATION_PLANE_OFFSET_KCPS);
+
+                return (ushort)((xTalk * 1000) >> 9); /* * 1000 to convert kcps to cps and >> 9 (7.9 format) */
+            }
+
+            set
+            {
+                WriteUInt16((ushort)Registers.ALGO__CROSSTALK_COMPENSATION_X_PLANE_GRADIENT_KCPS, 0x0000);
+                WriteUInt16((ushort)Registers.ALGO__CROSSTALK_COMPENSATION_Y_PLANE_GRADIENT_KCPS, 0x0000);
+                WriteUInt16((ushort)Registers.ALGO__CROSSTALK_COMPENSATION_PLANE_OFFSET_KCPS,
+                    (ushort)((value << 9) / 1000)); /* * << 9 (7.9 format) and /1000 to convert cps to kpcs */
+            }
+        }
+
+        /// <summary>
+        /// This function programs the threshold detection mode.
+        /// For example:
+        /// SetDistanceThreshold(dev,100,300, WindowDetectionMode.Below): below 100
+        /// SetDistanceThreshold(dev,100,300, WindowDetectionMode.Above): above 300
+        /// SetDistanceThreshold(dev,100,300, WindowDetectionMode.Out): out-of-window
+        /// SetDistanceThreshold(dev,100,300, WindowDetectionMode.In): in window
+        /// </summary>
+        /// <param name="threshLow">
+        /// The threshold under which the device raises an interrupt if detectionMode =
+        /// WindowDetectionMode.Below
+        /// </param>
+        /// <param name="threshHigh">
+        /// The threshold above which the device raises an interrupt if detectionMode =
+        /// WindowDetectionMode.Above
+        /// </param>
+        /// <param name="detectionMode">The <see cref="WindowDetectionMode" /> where 0 = below, 1 = above, 2 = out, and 3 = in</param>
+        public void SetDistanceThreshold(Length threshLow, Length threshHigh, WindowDetectionMode detectionMode)
+        {
+            byte temp = ReadByte((ushort)Registers.SYSTEM__INTERRUPT_CONFIG_GPIO);
+            temp &= 0x47;
+            WriteRegister((ushort)Registers.SYSTEM__INTERRUPT_CONFIG_GPIO,
+                (byte)(temp | ((byte)detectionMode & 0x07) | 0x40));
+            WriteUInt16((ushort)Registers.SYSTEM__THRESH_HIGH, (ushort)threshHigh.Millimeters);
+            WriteUInt16((ushort)Registers.SYSTEM__THRESH_LOW, (ushort)threshLow.Millimeters);
+        }
+
+        /// <summary>
+        /// Returns the <see cref="WindowDetectionMode" />.
+        /// </summary>
+        public WindowDetectionMode DistanceThresholdWindowDetectionMode
+        {
+            get
+            {
+                byte temp = ReadByte((ushort)Registers.SYSTEM__INTERRUPT_CONFIG_GPIO);
+                return
+                    (WindowDetectionMode)(temp &
+                                          0x7);
+            }
+        }
+
+        /// <summary>
+        /// Returns the low threshold.
```

Comment:
Would be good to add the typical range here as well

Reply:
Unfortunately. The api documentation does not provide a typical range for the parameter.